src/analyses/base.ml

```diff
@@ -2151,7 +2151,7 @@ struct
      | _ -> ()
     );
     match lval with (* this section ensure global variables contain bottom values of the proper type before setting them  *)
-    | (Var v, offs) when AD.is_definite lval_val && v.vglob ->
```

Comment:
There's an odd conundrum here, because as it turns out the all-index offsets used for `exp.fast_global_inits` are actually not considered definite (they evaluate to intdomain top), so all this logic that's supposed to ensure globals are initialized properly hasn't actually been applying there at all.
Now the problem is this:
1. 03-base/28-base-mutex-macos, which exposed the problem, requires it to happen (on MacOS the internal mutex structure contains an array), otherwise we get unsound invariants about the internals of the mutex that we want to hide.
2. 30-fast_global_inits tests, however, start failing because apparently partitioned arrays in some intricate way depend on the old accidental behavior that the initialization was skipped.

@michael-schwarz Maybe you have some insights into this? `ValueDomain.update_offset` does have a special case for initializing array from superbot, but somehow if `ValueDomain.bot_value` is used to do it first, then the behavior is different it seems.

Reply:
No, I also don't have any deeper insights here. I'll have to go through the code again.

Reply:
This check was independently removed in https://github.com/goblint/analyzer/pull/970/files#r1122731304 and is no longer part of the diff of this PR.